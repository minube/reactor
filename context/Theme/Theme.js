import { node, shape } from 'prop-types';
import React, { createContext, PureComponent } from 'react';

import THEME from '../../common/theme';

const { Provider, Consumer: ConsumerTheme } = createContext('reactor:l10n');

class ProviderTheme extends PureComponent {
  static propTypes = {
    children: node,
    style: shape({}),
  };

  static defaultProps = {
    children: undefined,
    style: {},
  };

  constructor(props) {
    super(props);
    this.state = { theme: THEME.extend(props.style) };
  }

  componentWillReceiveProps({ style: nextStyle }) {
    const { props: { style } } = this;
    if (nextStyle !== style) this.setState({ theme: THEME.extend(nextStyle) });
  }

  render() {
    const { props: { children }, state: { theme } } = this;

    return (
      <Provider value={{ theme }}>
        { children }
      </Provider>
    );
  }
}

export { ConsumerTheme, ProviderTheme };
