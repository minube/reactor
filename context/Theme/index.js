export { ConsumerTheme, ProviderTheme } from './Theme';
