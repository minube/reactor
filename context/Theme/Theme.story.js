import React, { Component, Fragment } from 'react';
import { View } from 'react-native';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { text, boolean, color, object } from '@storybook/addon-knobs/react';

import { Button, Text } from '../../components';
import { ConsumerTheme, ProviderTheme } from './Theme';


const style = {
  COLOR: { PRIMARY: 'red' },
  FONT: { FAMILY: 'Courier' },
};

class ThemeHOC extends Component {
  state = {
    night: false,
  };

  render() {
    const { _onChange, state: { night } } = this;

    return (
      <ProviderTheme style={night ? style : undefined}>
        <Fragment>
          <Button title="Switch Mode" onPress={() => this.setState({ night: !night })} />
          <Text>
            Hello World
          </Text>
        </Fragment>
      </ProviderTheme>
    );
  }
}

storiesOf('☑️ contextTheme', module)
  .add('default', () => (
    <ThemeHOC />
  ))
  // .add('style', () => (
  //   <ProviderTheme title="Press me" style={styles.button} onPress={_onPress} />
  // ))
  // .add('🏀 Playground', () => (
  //   <ProviderTheme
  //     secondary={boolean('secondary', false)}
  //     activity={boolean('activity', false)}
  //     color={color('color', undefined)}
  //     contained={boolean('contained', true)}
  //     disabled={boolean('disabled', false)}
  //     icon={text('icon', undefined)}
  //     onPress={action('Button.onPress()')}
  //     outlined={boolean('outlined', false)}
  //     primary={boolean('primary', false)}
  //     responsive={boolean('responsive', false)}
  //     rounded={boolean('rounded', false)}
  //     small={boolean('small', false)}
  //     title={text('title', undefined)}
  //     style={object('style', styles.button)}
  //   />
  // ));

