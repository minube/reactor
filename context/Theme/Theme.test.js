import React from 'react';
import TestRenderer from 'react-test-renderer';

import { ConsumerTheme, ProviderTheme } from './Theme';
import THEME from '../../common/theme';

const theme = {
  COLOR: { PRIMARY: 'green' },
};

describe('<ProviderTheme>', () => {
  it('renders', () => {
    const tree = TestRenderer.create(<ProviderTheme />).toJSON();
    expect(tree).toMatchSnapshot();
  });

   it('when {style}', () => {
    expect(THEME.COLOR.PRIMARY).toEqual('#bdbdbd');
    const tree = TestRenderer.create(<ProviderTheme style={theme} />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(THEME.COLOR.PRIMARY).toEqual(theme.COLOR.PRIMARY);
  });
});

