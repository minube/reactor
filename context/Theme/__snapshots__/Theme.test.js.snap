// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProviderTheme> renders 1`] = `null`;

exports[`<ProviderTheme> when {style} 1`] = `null`;
